packages/swapper/src/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -471,169 +147,101 @@ export const thorchainApi: SwapperApi = {
-              basePath: config.VITE_UNCHAINED_THORCHAIN_HTTP_URL,
-            }),
-          )
-
-          const account = await api.getAccount({ pubkey: from })
-
-          return { fee, msg, account }
-        }
-        case tcyAssetId: {
-          // Another day in Cosmos SDK land, yes TCY is a native asset, but TCY is the fee asset so we *do* need both
-          const fee: StdFee = {
-            amount: [{ amount: deductOutboundRuneFee(networkFee), denom: 'rune' }],
-            gas,
-          }
-
-          // https://dev.thorchain.org/thorchain-dev/concepts/memos#asset-notation
-          const msg: cosmossdkChainAdapter.ThorchainMsgDeposit = {
-            type: cosmossdkChainAdapter.ThorchainMessageType.MsgDeposit,
-            value: {
-              coins: [{ asset: 'THOR.TCY', amount: sellAmountIncludingProtocolFeesCryptoBaseUnit }],
-              memo,
-              signer: from,
-            },
-          }
-
-          const api = new cosmossdk.thorchain.V1Api(
-            new cosmossdk.thorchain.Configuration({
-              basePath: config.VITE_UNCHAINED_THORCHAIN_HTTP_URL,
-            }),
-          )
-
-          const account = await api.getAccount({ pubkey: from })
-
-          return { fee, msg, account }
-        }
+    const fee = feeData.networkFeeCryptoBaseUnit ?? '0'
 
-        case cosmosAssetId: {
-          const fee: StdFee = {
-            amount: [{ amount: networkFee, denom: 'uatom' }],
-            gas,
-          }
-
-          const daemonUrl = config.VITE_THORCHAIN_NODE_URL
-          const maybeGaiaAddressData = await getInboundAddressDataForChain(daemonUrl, cosmosAssetId)
-          if (maybeGaiaAddressData.isErr()) throw maybeGaiaAddressData.unwrapErr()
-          const gaiaAddressData = maybeGaiaAddressData.unwrap()
-          const vault = gaiaAddressData.address
-
-          const msg: cosmossdkChainAdapter.CosmosSdkMsgSend = {
-            type: cosmossdkChainAdapter.CosmosSdkMessageType.MsgSend,
-            value: {
-              amount: [{ amount: sellAmountIncludingProtocolFeesCryptoBaseUnit, denom: 'uatom' }],
-              from_address: from,
-              to_address: vault,
-            },
-          }
-
-          const api = new cosmossdk.cosmos.V1Api(
-            new cosmossdk.cosmos.Configuration({
-              basePath: config.VITE_UNCHAINED_COSMOS_HTTP_URL,
-            }),
-          )
-
-          const account = await api.getAccount({ pubkey: from })
-
-          return { fee, msg, account }
-        }
+    switch (sellAsset.assetId) {
+      case thorchainAssetId:
+      case tcyAssetId: {
+        const adapter = assertGetCosmosSdkChainAdapter(sellAsset.chainId) as thorchain.ChainAdapter
+
+        const { txToSign } = await adapter.buildDepositTransaction({
+          accountNumber,
+          from,
+          value: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+          memo,
+          chainSpecific: { gas, fee, coin: tcyAssetId ? 'THOR.TCY' : 'THOR.RUNE' },
```

Comment:
This is wrong, every trades from RUNE will be using TCY with this as tcyAssetId is a constant it will always fall in the true case

Reply:
oh my, good catch! 05889bfc70b16a1c900974a38299fa391f528f61